x-pack/plugins/case/server/client/cases/update.ts

```diff
@@ -291,33 +291,25 @@ async function updateAlerts({
   // get a map of sub case id to the sub case status
   const subCasesToStatus = await getSubCasesToStatus({ totalAlerts, client, caseService });
 
-  // create a map of the case statuses to the alert information that we need to update for that status
-  // This allows us to make at most 3 calls to ES, one for each status type that we need to update
-  // One potential improvement here is to do a tick (set timeout) to reduce the memory footprint if that becomes an issue
-  const alertsToUpdate = totalAlerts.saved_objects.reduce((acc, alertComment) => {
-    if (isCommentRequestTypeAlertOrGenAlert(alertComment.attributes)) {
-      const status = getSyncStatusForComment({
-        alertComment,
-        casesToSyncToStatus,
-        subCasesToStatus,
-      });
+  // create an array of requests that indicate the id, index, and status to update an alert
+  const alertsToUpdate = totalAlerts.saved_objects.reduce(
+    (acc: UpdateAlertRequest[], alertComment) => {
+      if (isCommentRequestTypeAlertOrGenAlert(alertComment.attributes)) {
```

Comment:
Not necessary for this PR, but just thinking it may be better to have functions that only care about one status, i.e. `isCommentRequestType(attributes) || isGenAlert(attributes)`. I just imagine other forms of alerts in the near or far future appearing and this becoming more unwieldy.